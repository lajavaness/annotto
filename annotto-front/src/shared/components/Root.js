import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { configure } from 'react-hotkeys'
import { createStore } from 'redux-dynamic-modules'
import { getSagaExtension } from 'redux-dynamic-modules-saga'
import React from 'react'

import rootModule from 'modules/root'

import sagas from 'shared/sagas'

import theme from '__theme__'

import App from 'shared/components/app/App'

import 'antd/dist/antd.less'
import 'assets/locales'

const sagaExtension = getSagaExtension()

const store = createStore({
	extensions: [sagaExtension],
})

sagaExtension.middleware[0].run(sagas)

store.addModule(rootModule)

configure({
	ignoreEventsCondition: (event) => {
		const { target } = event

		if (target && target.tagName) {
			const tagName = target.tagName.toLowerCase()
			const type = target.getAttribute('type')
			if (
				['select', 'textarea'].includes(tagName) ||
				(tagName === 'input' && type && !['checkbox', 'radio'].includes(type))
			) {
				return true
			}
		}

		return false
	},
})

const Root = () => (
	<Provider store={store}>
		<ThemeProvider theme={theme}>
			<App />
		</ThemeProvider>
	</Provider>
)

export default Root
