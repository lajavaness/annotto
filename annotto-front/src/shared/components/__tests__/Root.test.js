import React from 'react'

import { render } from '@testing-library/react'

import Root from '../Root'

const getInstance = () => <Root />

describe('Root', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
