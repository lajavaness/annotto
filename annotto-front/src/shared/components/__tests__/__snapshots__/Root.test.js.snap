// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe obZLm"
    data-testid="__loader__"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
